x/axelarnet/keeper/msg_server.go

```diff
@@ -80,11 +80,24 @@ func (s msgServer) CallContract(c context.Context, req *types.CallContractReques
 	payloadHash := crypto.Keccak256(req.Payload)
 
 	msgID, txID, nonce := s.nexus.GenerateMessageID(ctx)
-	msg := nexus.NewGeneralMessage(msgID, sender, recipient, payloadHash, nexus.Processing, txID, nonce, nil)
+	msg := nexus.NewGeneralMessage(msgID, sender, recipient, payloadHash, nexus.Approved, txID, nonce, nil)
 	if err := s.nexus.SetNewMessage(ctx, msg); err != nil {
 		return nil, sdkerrors.Wrap(err, "failed to add general message")
 	}
 
+	if req.Fee != nil {
+		events.Emit(ctx, &types.FeePaid{
+			MessageID: msgID,
+			Recipient: req.Fee.Recipient,
+			Fee:       req.Fee.Amount,
+		})
+
+		err := s.bank.SendCoins(ctx, req.Sender, req.Fee.Recipient, sdk.NewCoins(req.Fee.Amount))
```

Comment:
Should we restrict the fee payment denom to be an axelar registered asset? We can to be consistent, not like the user will use other denoms on axelar anyways, but i'm also ok with not restricting

Reply:
If we choose to restrict the fee denom, then we should also restrict the recipient. It makes no sense to restrict one but other the other.

Reply:
I think it's simpler to just not restrict